### PR TITLE
capitalized all [[LocaleRelated]] slots

### PR DESCRIPTION
**Pull request by ben-allen**

### CODE REVIEW

**Review by ljharb — APPROVED** [review]

File: spec/datetimeformat.html [hunk 1]
```diff
@@ -223,13 +223,13 @@ <h1>Internal slots</h1>
           </ul>
           Each of the records must also have the following fields:
```
Comment (gibson042) [inline spec/datetimeformat.html:224]:
```suggestion
          Each Date-Time Format Pattern Record must also have the following fields:
```

File: spec/datetimeformat.html [hunk 2]
```diff
@@ -239,37 +239,37 @@ <h1>Internal slots</h1>
...
+          For [[DateTimeFormat]], the field value must be a string pattern which contains the strings *"{0}"* and *"{1}"*. For [[DateTimeRangeFormat]] the value of these fields must be a nested record which also has [[Full]], [[Long]], [[Medium]] and [[Short]] fields. The [[Full]], [[Long]], [[Medium]] and [[Short]] fields in the enclosing record refer to the date style of the range pattern, while the fields in the nested record refers to the time style of the range pattern. The value of these fields in the nested record is a record with a [[RangePatterns]] field and a [[RangePatterns12]] field which are similar to the [[RangePatterns]] and [rangePatterns12]] fields in [[DateFormat]] and [[TimeFormat]].
          </li>
```
Comment (gibson042) [inline spec/datetimeformat.html:258]: These [[full]]/[[long]]/etc. fields also must be lowercase to correspond with the current definition of [DateTimeStyleFormat](https://tc39.es/ecma402/#sec-date-time-style-format).
```suggestion
        <li>
          [[LocaleData]].[[&lt;_locale_&gt;]] must have a [[Styles]] field. This [[Styles]] field must be a Record with [[&lt;_calendar_&gt;]] fields for all calendar values _calendar_. The value of each [[&lt;_calendar_&gt;]] field must be a Record with all of the following fields:
          <ul>
            <li>A [[DateFormat]] field whose value is a Record with [[full]], [[long]], [[medium]] and [[short]] fields, the value for each of which has the same structure and constraints as a Date-Time Format Pattern Record element of the [[LocaleData]].[[<_locale_>]].[[<_calendar_>]] List.</li>
            <li>A [[TimeFormat]] field whose value is a Record with [[full]], [[long]], [[medium]] and [[short]] fields, the value for each of which has the same structure and constraints as a Date-Time Format Pattern Record element of the [[LocaleData]].[[<_locale_>]].[[<_calendar_>]] List.</li>
            <li>A [[DateTimeFormat]] field whose value is a Record with [[full]], [[long]], [[medium]] and [[short]] fields, the value for each of which is a string containing substrings *"{0}"* and *"{1}"* (serving as respective placeholders for time and date patterns).</li>
            <li>A [[DateTimeRangeFormat]] field whose value is a Record with [[full]], [[long]], [[medium]] and [[short]] fields (corresponding with the respective DateTimeFormat [[DateStyle]] values), the value for each of which is a Record describing a collection of format patterns for that date style. That nested Record has [[full]], [[long]], [[medium]] and [[short]] fields (corresponding with the respective DateTimeFormat [[TimeStyle]] values), the value for each of which is a Record with [[RangePatterns]] and [[RangePatterns12]] fields whose values have the same structure and constraints as the respectively-named fields of a Date-Time Format Pattern Record.</li>
          </ul>
        </li>
```

File: spec/datetimeformat.html [hunk 3]
```diff
@@ -239,37 +239,37 @@ <h1>Internal slots</h1>
...
-          <li>[[hour]]: *"numeric"*</li>
-          <li>[[minute]]: *"numeric"*</li>
-          <li>[[pattern]]: *"{hour}:{minute}"*</li>
-          <li>[[pattern12]]: *"{hour}:{minute} {ampm}"*</li>
-          <li>[[rangePatterns]]:</li>
+          <li>[[Hour]]: *"numeric"*</li>
+          <li>[[Minute]]: *"numeric"*</li>
+          <li>[[Pattern]]: *"{hour}:{minute}"*</li>
+          <li>[[Pattern12]]: *"{hour}:{minute} {ampm}"*</li>
+          <li>[[RangePatterns]]:</li>
```
Comment (gibson042) [inline spec/datetimeformat.html:268]:
```suggestion
          <li>[[hour]]: *"numeric"*</li>
          <li>[[minute]]: *"numeric"*</li>
          <li>[[Pattern]]: *"{hour}:{minute}"*</li>
          <li>[[Pattern12]]: *"{hour}:{minute} {ampm}"*</li>
          <li>[[RangePatterns]]:</li>
```

File: spec/datetimeformat.html [hunk 4]
```diff
@@ -239,37 +239,37 @@ <h1>Internal slots</h1>
...
           <ul>
             <li>[[Hour]]:<ul>
-              <li>[[hour]]: *"numeric"*</li>
-              <li>[[minute]]: *"numeric"*</li>
+              <li>[[Hour]]: *"numeric"*</li>
+              <li>[[Minute]]: *"numeric"*</li>
```
Comment (gibson042) [inline spec/datetimeformat.html:272]:
```suggestion
              <li>[[hour]]: *"numeric"*</li>
              <li>[[minute]]: *"numeric"*</li>
```

File: spec/datetimeformat.html [hunk 5]
```diff
@@ -278,8 +278,8 @@ <h1>Internal slots</h1>
               </ul>
             </ul></li>
             <li>[[Minute]]:<ul>
-              <li>[[hour]]: *"numeric"*</li>
-              <li>[[minute]]: *"numeric"*</li>
+              <li>[[Hour]]: *"numeric"*</li>
+              <li>[[Minute]]: *"numeric"*</li>
```
Comment (gibson042) [inline spec/datetimeformat.html:282]:
```suggestion
              <li>[[hour]]: *"numeric"*</li>
              <li>[[minute]]: *"numeric"*</li>
```

File: spec/datetimeformat.html [hunk 6]
```diff
@@ -288,11 +288,11 @@ <h1>Internal slots</h1>
...
-              <li>[[year]]: *"2-digit"*</li>
-              <li>[[month]]: *"numeric"*</li>
-              <li>[[day]]: *"numeric"*</li>
-              <li>[[hour]]: *"numeric"*</li>
-              <li>[[minute]]: *"numeric"*</li>
+              <li>[[Year]]: *"2-digit"*</li>
+              <li>[[Month]]: *"numeric"*</li>
+              <li>[[Day]]: *"numeric"*</li>
+              <li>[[Hour]]: *"numeric"*</li>
+              <li>[[Minute]]: *"numeric"*</li>
```
Comment (gibson042) [inline spec/datetimeformat.html:295]:
```suggestion
              <li>[[year]]: *"2-digit"*</li>
              <li>[[month]]: *"numeric"*</li>
              <li>[[day]]: *"numeric"*</li>
              <li>[[hour]]: *"numeric"*</li>
              <li>[[minute]]: *"numeric"*</li>
```

File: spec/datetimeformat.html [hunk 7]
```diff
@@ -301,11 +301,11 @@ <h1>Internal slots</h1>
...
             </ul></li>
           </ul>
-          <li>[[rangePatterns12]]:
+          <li>[[RangePatterns12]]:
           <ul>
             <li>[[Hour]]:<ul>
-              <li>[[hour]]: *"numeric"*</li>
-              <li>[[minute]]: *"numeric"*</li>
+              <li>[[Hour]]: *"numeric"*</li>
+              <li>[[Minute]]: *"numeric"*</li>
```
Comment (gibson042) [inline spec/datetimeformat.html:308]:
```suggestion
              <li>[[hour]]: *"numeric"*</li>
              <li>[[minute]]: *"numeric"*</li>
```

File: spec/datetimeformat.html [hunk 8]
```diff
@@ -315,8 +315,8 @@ <h1>Internal slots</h1>
               </ul>
             </ul></li>
             <li>[[Minute]]:<ul>
-              <li>[[hour]]: *"numeric"*</li>
-              <li>[[minute]]: *"numeric"*</li>
+              <li>[[Hour]]: *"numeric"*</li>
+              <li>[[Minute]]: *"numeric"*</li>
```
Comment (gibson042) [inline spec/datetimeformat.html:319]:
```suggestion
              <li>[[hour]]: *"numeric"*</li>
              <li>[[minute]]: *"numeric"*</li>
```

File: spec/datetimeformat.html [hunk 9]
```diff
@@ -326,11 +326,11 @@ <h1>Internal slots</h1>
...
-              <li>[[year]]: *"2-digit"*</li>
-              <li>[[month]]: *"numeric"*</li>
-              <li>[[day]]: *"numeric"*</li>
-              <li>[[hour]]: *"numeric"*</li>
-              <li>[[minute]]: *"numeric"*</li>
+              <li>[[Year]]: *"2-digit"*</li>
+              <li>[[Month]]: *"numeric"*</li>
+              <li>[[Day]]: *"numeric"*</li>
+              <li>[[Hour]]: *"numeric"*</li>
+              <li>[[Minute]]: *"numeric"*</li>
```
Comment (gibson042) [inline spec/datetimeformat.html:333]:
```suggestion
              <li>[[year]]: *"2-digit"*</li>
              <li>[[month]]: *"numeric"*</li>
              <li>[[day]]: *"numeric"*</li>
              <li>[[hour]]: *"numeric"*</li>
              <li>[[minute]]: *"numeric"*</li>
```

File: spec/listformat.html [hunk 10]
```diff
@@ -91,7 +91,7 @@ <h1>Internal slots</h1>
       </p>
 
       <ul>
-        <li>[[LocaleData]].[[&lt;_locale_&gt;]] is a Record which has three fields [[conjunction]], [[disjunction]], and [[unit]]. Each of these is a Record which must have fields with the names of three formatting styles: [[long]], [[short]], and [[narrow]].</li>
+        <li>[[LocaleData]].[[&lt;_locale_&gt;]] is a Record which has three fields [[Conjunction]], [[Disjunction]], and [[Unit]]. Each of these is a Record which must have fields with the names of three formatting styles: [[Long]], [[Short]], and [[Narrow]].</li>
```
Comment (gibson042) [inline spec/listformat.html:94]: These also cannot change while the [Intl.ListFormat constructor](https://tc39.es/ecma402/#sec-Intl.ListFormat) uses lowercase option values as lookup keys.
```suggestion
        <li>[[LocaleData]].[[&lt;_locale_&gt;]] is a Record which has three fields [[conjunction]], [[disjunction]], and [[unit]]. Each of these is a Record which must have fields with the names of three formatting styles: [[long]], [[short]], and [[narrow]].</li>
```

File: spec/numberformat.html [hunk 11]
```diff
@@ -239,7 +239,7 @@ <h1>Internal slots</h1>
 
       <ul>
         <li>The list that is the value of the *"nu"* field of any locale field of [[LocaleData]] must not include the values *"native"*, *"traditio"*, or *"finance"*.</li>
-        <li>[[LocaleData]].[[&lt;_locale_&gt;]] must have a [[patterns]] field for all locale values _locale_. The value of this field must be a Record, which must have fields with the names of the four number format styles: *"decimal"*, *"percent"*, *"currency"*, and *"unit"*.</li>
+        <li>[[LocaleData]].[[&lt;_locale_&gt;]] must have a [[Patterns]] field for all locale values _locale_. The value of this field must be a Record, which must have fields with the names of the four number format styles: *"decimal"*, *"percent"*, *"currency"*, and *"unit"*.</li>
```
Comment (gibson042) [inline spec/numberformat.html:242]: This change _can_ work, but it requires some finesse.
```suggestion
        <li>[[LocaleData]].[[&lt;_locale_&gt;]] must have a [[Patterns]] field for all locale values _locale_. The value of this field must be a Record, which must have fields corresponding with  the names of the four number format styles after uppercasing the initial letter: [[Decimal]], [[Percent]], [[Currency]], and [[Unit]].</li>
```

File: spec/numberformat.html [hunk 12]
```diff
@@ -239,7 +239,7 @@ <h1>Internal slots</h1>
...
         <li>
           The two fields *"currency"* and *"unit"* noted above must be Records with at least one field, *"fallback"*.
           The *"currency"* may have additional fields with keys corresponding to currency codes according to <emu-xref href="#sec-currency-codes"></emu-xref>.
```
Comment (gibson042) [inline spec/numberformat.html:245]:
```suggestion
          The two fields [[Currency]] and [[Unit]] noted above must be Records with at least one field, [[Fallback]].
          The [[Currency]] Record may have additional fields with keys corresponding to currency codes according to <emu-xref href="#sec-currency-codes"></emu-xref>.
```
and so on. Note also that the change from "fallback" to "Fallback" will require fixes in [GetNumberFormatPattern](https://tc39.es/ecma402/#sec-getnumberformatpattern).

File: spec/numberformat.html [hunk 13]
```diff
@@ -254,7 +254,7 @@ <h1>Internal slots</h1>
           Additionally, the values within the *"percent"* field must also contain the substring *"{percentSign}"*; the values within the *"currency"* field must also contain one or more of the following substrings: *"{currencyCode}"*, *"{currencyPrefix}"*, or *"{currencySuffix}"*; and the values within the *"unit"* field must also contain one or more of the following substrings: *"{unitPrefix}"* or *"{unitSuffix}"*.
```
Comment (gibson042) [inline spec/numberformat.html:254]:
```suggestion
          Additionally, the values within the [[Percent]] field must also contain the <emu-not-ref>substring</emu-not-ref> *"{percentSign}"*; the values within the [[Currency]] field must also contain one or more of the following substrings: *"{currencyCode}"*, *"{currencyPrefix}"*, or *"{currencySuffix}"*; and the values within the [[Unit]] field must also contain one or more of the following substrings: *"{unitPrefix}"* or *"{unitSuffix}"*.
```

File: spec/numberformat.html [hunk 14]
```diff
@@ -1132,15 +1132,15 @@ <h1>GetNumberFormatPattern ( _numberFormat_, _x_ )</h1>
...
-          1. Let _patterns_ be _patterns_.[[percent]].
+          1. Let _patterns_ be _patterns_.[[Percent]].
         1. Else if _style_ is *"unit"*, then
           1. Let _unit_ be _numberFormat_.[[Unit]].
           1. Let _unitDisplay_ be _numberFormat_.[[UnitDisplay]].
-          1. Let _patterns_ be _patterns_.[[unit]].
+          1. Let _patterns_ be _patterns_.[[Unit]].
           1. If _patterns_ doesn't have a field [[&lt;_unit_&gt;]], then
             1. Let _unit_ be *"fallback"*.
           1. Let _patterns_ be _patterns_.[[&lt;_unit_&gt;]].
```
Comment (gibson042) [inline spec/numberformat.html:1146]:
```suggestion
          1. If _patterns_ doesn't have a field [[&lt;_unit_&gt;]], then
            1. Let _patterns_ be _patterns_.[[Fallback]].
          1. Else,
            1. Let _patterns_ be _patterns_.[[&lt;_unit_&gt;]].
```

File: spec/numberformat.html [hunk 15]
```diff
@@ -1149,36 +1149,36 @@ <h1>GetNumberFormatPattern ( _numberFormat_, _x_ )</h1>
           1. Let _currency_ be _numberFormat_.[[Currency]].
           1. Let _currencyDisplay_ be _numberFormat_.[[CurrencyDisplay]].
           1. Let _currencySign_ be _numberFormat_.[[CurrencySign]].
-          1. Let _patterns_ be _patterns_.[[currency]].
+          1. Let _patterns_ be _patterns_.[[Currency]].
           1. If _patterns_ doesn't have a field [[&lt;_currency_&gt;]], then
             1. Let _currency_ be *"fallback"*.
           1. Let _patterns_ be _patterns_.[[&lt;_currency_&gt;]].
           1. Let _patterns_ be _patterns_.[[&lt;_currencyDisplay_&gt;]].
           1. Let _patterns_ be _patterns_.[[&lt;_currencySign_&gt;]].
```
Comment (gibson042) [inline spec/numberformat.html:1157]:
```suggestion
          1. If _patterns_ doesn't have a field [[&lt;_currency_&gt;]], then
            1. Let _patterns_ be _patterns_.[[Fallback]].
          1. Else,
            1. Let _patterns_ be _patterns_.[[&lt;_currency_&gt;]].
          1. Let _patterns_ be _patterns_.[[&lt;_currencyDisplay_&gt;]].
          1. Let _patterns_ be _patterns_.[[&lt;_currencySign_&gt;]].
```

**Review by gibson042 — COMMENTED** [review]
Thank you! But unfortunately, widespread use of the [[[\<_name_>]]](https://tc39.es/ecma402/#conventions) indirection pattern to map option values directly to field names makes it much more complicated than a straightforward find and replace.

**Review by gibson042 — CHANGES_REQUESTED** [review]
As noted above, this needs tweaking.

**Pull request closed without merging**